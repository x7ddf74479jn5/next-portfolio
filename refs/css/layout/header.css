.l-header {
  transition: top 0.4s ease;
  background-color: #fff;
  box-sizing: content-box;
  border-bottom: 1px solid rgba(0, 0, 0, 0.2);
  box-shadow: 0 2px 1px rgba(0, 0, 0, 0.1);
  position: relative;
  z-index: 100;
}
.l-header .l-header-inner {
  padding-right: 32px;
  margin: 0 auto;
}
.l-header .l-product-logo {
  padding: 0 1rem;
  background: #a2dbf3;
  display: none;
  margin: 0;
  overflow: hidden;
  position: relative;
  text-overflow: ellipsis;
  vertical-align: top;
  white-space: nowrap;
}
@media screen and (min-width: 1024px) {
  .l-header .l-product-logo {
    background: transparent;
    display: block;
    float: left;
  }
  .l-header .l-product-logo a {
    height: 64px;
    width: 128px;
  }
  .l-header .l-product-logo a img, .l-header .l-product-logo a svg {
    height: auto;
    width: 128px;
  }
}
.l-header .nav-toggle-button {
  background: 0 0;
  border: 0;
  color: transparent;
  cursor: pointer;
  height: 56px;
  right: 0;
  line-height: 3.75rem;
  outline: 0;
  padding: 0;
  position: absolute;
  text-align: center;
  top: 0;
  width: 3.75rem;
}
.l-header .nav-toggle-button img, .l-header .nav-toggle-button svg {
  color: #757575;
  height: 1.5rem;
  margin: 0 auto;
  width: 1.5rem;
}
.l-header .nav-toggle-button img.menu-bars, .l-header .nav-toggle-button svg.menu-bars {
  display: block;
}
.l-header .nav-toggle-button img.menu-close, .l-header .nav-toggle-button svg.menu-close {
  display: none;
}
@media screen and (min-width: 1024px) {
  .l-header .nav-toggle-button {
    display: none;
  }
}
.l-header.fixed-nav {
  overflow: visible;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 2000;
}
.l-header__nav-top {
  right: -86%;
  opacity: 0;
  transition: right 1.2s cubic-bezier(0.19, 1, 0.22, 1);
  overflow: hidden;
}
.l-header__nav-top .l-product-logo {
  display: none;
}
.l-header__nav-top ul {
  display: none;
  margin-left: 1.1rem;
}
.l-header__nav-top a {
  color: #4c4c4c;
  font-weight: 300;
  letter-spacing: 0.01562rem;
  line-height: 1.375rem;
  display: inline-block;
  padding: 21px 0 19px;
}
.l-header__nav-top a:hover {
  color: #777;
}
.l-header__nav-top .sns img {
  height: 2rem;
  width: 2rem;
}
@media screen and (min-width: 1024px) {
  .l-header__nav-top {
    display: block;
    opacity: 1;
    overflow: visible;
  }
  .l-header__nav-top ul {
    background: 0 0;
    display: block;
    float: right;
    margin-bottom: 0;
    position: relative;
    top: 1px;
  }
  .l-header__nav-top li {
    display: inline-block;
    margin: 0 1rem;
  }
  .l-header__nav-top li:first-child {
    margin-left: 0;
  }
  .l-header__nav-top .sns span {
    display: none;
  }
}
.nav-open {
  height: 100%;
  margin: 0;
  overflow: hidden;
  padding: 0;
}
.nav-open main {
  overflow: hidden;
  pointer-events: none;
  -ms-touch-action: none;
  touch-action: none;
}
.nav-open .l-header {
  overflow: visible;
}
.nav-open .l-header .nav-toggle-button {
  left: auto;
  right: 0.3125rem;
  z-index: 1200;
}
.nav-open .l-header .nav-toggle-button img.menu-bars, .nav-open .l-header .nav-toggle-button svg.menu-bars {
  display: none;
}
.nav-open .l-header .nav-toggle-button img.menu-close, .nav-open .l-header .nav-toggle-button svg.menu-close {
  display: block;
}
.nav-open .l-header__nav-top {
  background: #fff;
  display: block;
  height: 100%;
  right: 0;
  opacity: 1;
  position: fixed;
  top: 0;
  max-width: 25rem;
  width: 86%;
  z-index: 1100;
}
.nav-open .l-header__nav-top .l-product-logo {
  display: block;
  padding: 4px 30px 0;
  z-index: 1200;
}
.nav-open .l-header__nav-top .l-product-logo a {
  border-bottom: 0;
  height: 40px;
}
.nav-open .l-header__nav-top .l-product-logo img, .nav-open .l-header__nav-top .l-product-logo svg {
  position: absolute;
  top: 4px;
  left: 1.1rem;
  height: 40px;
  width: auto;
}
.nav-open .l-header__nav-top ul {
  display: block;
  margin: 0;
}
.nav-open .l-header__nav-top ul li a {
  border-bottom: 0;
  display: block;
  padding: 13px 24px;
}
.nav-open .black-bg {
  background: rgba(0, 0, 0, .1);
  height: 100vh;
  position: fixed;
  right: 0;
  width: 100vw;
  z-index: 1000;
}
