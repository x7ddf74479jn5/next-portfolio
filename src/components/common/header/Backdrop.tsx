import styles from "src/styles/components/header/backdrop.module.scss";

type Props = {
  handleClick: () => void;
};

const Backdrop: React.VFC<Props> = ({ handleClick }) => {
  return (
    <button
      className={styles.backdrop}
      id="js-black-bg"
      onKeyDown={(e) => {
        return;
      }}
      onClick={() => handleClick()}
    ></button>
  );
};

export default Backdrop;
