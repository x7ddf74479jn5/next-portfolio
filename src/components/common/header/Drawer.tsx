import React from "react";
import Modal from "react-modal";
import "./styles.css";

Modal.setAppElement("#__next");

export default function App() {
  const [modalIsOpen, setIsOpen] = React.useState<boolean>(false);

  return (
    <div className="App">
      <button onClick={() => setIsOpen(true)}>Open Modal</button>
      <Modal isOpen={modalIsOpen}>
        <button onClick={() => setIsOpen(false)}>Close Modal</button>
      </Modal>
    </div>
  );
}
