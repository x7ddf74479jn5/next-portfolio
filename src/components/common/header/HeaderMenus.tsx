import Image from "next/image";
import Link from "next/link";
import { images, links as l } from "src/utils/paths";

const links = [
  { label: "About", href: "/about/" },
  { label: "Samples", href: "/samples/" },
  { label: "Blog", href: "/" },
  { label: "Contact", href: "/contact/" },
];

type HeaderMenusProps = {
  closeNav(): void;
  // isDialogOpen: boolean;
  closeDialog(): void;
  openDialog(e?: React.MouseEvent<HTMLAnchorElement, MouseEvent>): void;
};

export const HeaderMenus: React.FC<HeaderMenusProps> = ({ closeDialog, openDialog, closeNav }) => {
  const content = links.map((link, index) => {
    return (
      <li key={index}>
        <Link href={link.href}>
          <a target="_self">{link.label}</a>
        </Link>
      </li>
    );
  });

  const handleClick = () => {
    //ボタンが押された時の処理
    console.log("opne");
    openDialog();
    closeNav();
  };

  const handleKeyUp = (e: any) => {
    if (e.code === "Enter" || e.code === "Space") {
      // openChatBot()
      console.log("open");
      openDialog();
      closeNav();
    }
  };

  // function handleKeyUp() {
  //   if (押されたのはENTERキーかSPACEキー?) {
  //     if (ボタンにフォーカスしている?) {
  //       //ボタンが押された時の処理
  //     }
  // }

  return (
    <nav className="l-header__nav-top">
      <h1 className="l-product-logo">
        <Link href={"/"}>
          <a target="_self">
            <Image height={80} width={40} alt="パンダシャーク" src={images.pandashark_rec} />
            {/* <img alt="パンダシャーク" src={images.pandashark_rec} /> */}
          </a>
        </Link>
        {/* <a href="/" target="_self">
          <img alt="パンダシャーク" src="/img/icons/logo.png" />
        </a> */}
      </h1>
      <ul>
        {content}
        <li>
          <a className="sns" href={l.twitter} target="_self">
            <img src="/img/icons/twitter_logo.png" alt="twitter" />
            <span>Twitter</span>
          </a>
        </li>
        <li>
          <a
            className="sns"
            tabIndex={0}
            role="button"
            onClick={() => {
              return handleClick();
            }}
          >
            <img src={images.robot} alt="chat-bot" />
            <span>Chat Bot</span>
          </a>
        </li>
        {/* <li>
          <button onClick={() => handleClick()}>
            <img src="/img/icons/twitter_logo.png" alt="twitter" />
            <span>Chat Bot</span>
          </button>
        </li> */}
        {/* <li>
        <a className="sns" href="#" target="_self">
          <img src="/img/icons/youtube_logo.png" />
          <span>YouTubeチャンネル</span>
        </a>
      </li> */}
      </ul>
    </nav>
  );
};
