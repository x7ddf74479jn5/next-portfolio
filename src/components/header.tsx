import Link from "next/link";
import * as React from "react";

const items = [
  { href: "/", label: "Home" },
  { href: "/about", label: "About" },
];

export const Header: React.FC = () => {
  const [open, setOpen] = React.useState(false);
  // const classes= {
  //   navOpen:"nav-open"
  // }
  const toggleNav = () => {
    setOpen((prevState) => !prevState);
  };
  const closeNav = () => {
    setOpen(false);
  };

  const navOpen = open ? "nav-open" : "";
  return (
    <div className={navOpen}>
      <header className={navOpen + " fixed-nav " + "l-header "}>
        {/* <h1>Title</h1>
    <nav>
      {items.map((item) => (
        <Link key={item.href} href={item.href}>
          <a style={{ display: "inline-block", padding: 12 }}>{item.label}</a>
        </Link>
      ))}
    </nav> */}

        <div className="l-header-inner">
          <h1 className="l-product-logo">
            <a href="/" target="_self" aria-label="パンダシャーク">
              <img alt="パンダシャーク" src="/img/icons/pandashark_logo.png" />
            </a>
          </h1>
          {/* <!-- Navigation toggle button for Smartphone--> */}
          <button
            className={navOpen + " outside " + "nav-toggle-button "}
            data-about-nav-toggle
            aria-label="ナビゲーション メニューの切り替え"
            id="nav-toggle-button"
            onClick={() => toggleNav()}
          >
            <img
              alt="ナビゲーション メニューを開く"
              className={navOpen + " menu-bars "}
              src="/img/icons/menu.png"
              height="24"
              width="24"
            />
            <img
              alt="ナビゲーション メニューを閉じる"
              className={navOpen + " menu-close "}
              src="/img/icons/close.png"
              height="24"
              width="24"
            />
          </button>
          <nav className="l-header__nav-top">
            <h1 className="l-product-logo">
              <a href="/" target="_self">
                <img alt="パンダシャーク" src="/img/icons/logo.png" />
              </a>
            </h1>
            <ul>
              <li>
                <a href="/about/" target="_self">
                  About
                </a>
              </li>
              <li>
                <a href="/samples/" target="_self">
                  Samples
                </a>
              </li>
              <li>
                <a href="/" target="_blank">
                  Blog
                </a>
              </li>
              <li>
                <a href="/contact/" target="_self">
                  Contact
                </a>
              </li>
              <li>
                <a className="sns" href="/" target="_self">
                  <img src="/img/icons/twitter_logo.png" alt="twitter" />
                  <span>Twitter</span>
                </a>
              </li>
              {/* <li>
                <a className="sns" href="#" target="_self">
                  <img src="/img/icons/youtube_logo.png" />
                  <span>YouTubeチャンネル</span>
                </a>
              </li> */}
            </ul>
          </nav>
          {/* <button>
          <div className="black-bg" role="button" id="js-black-bg" onClick={() => closeNav()}></div>
          </button> */}
        </div>
      </header>
    </div>
  );
};
