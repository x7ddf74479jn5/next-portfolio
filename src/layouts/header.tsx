import * as React from "react";
// import styles from "src/styles/components/common/PandasharkLogo.module.scss";

type HeaderProps = {
  open: boolean;
  isDialogOpen: boolean;
  toggleNav(): void;
  closeNav(e?: React.KeyboardEvent<HTMLDivElement>): void;
  closeDialog(): void;
  openDialog(e?: React.MouseEvent<HTMLAnchorElement, MouseEvent>): void;
};

export const Header: React.VFC<HeaderProps> = ({
  open,
  isDialogOpen,
  toggleNav,
  closeNav,
  closeDialog,
  openDialog,
}) => {
  // const [open, setOpen] = React.useState(false);
  // const classes= {
  //   navOpen:"nav-open"
  // }
  // const toggleNav = () => {
  //   setOpen((prevState) => !prevState);
  // };
  // const closeNav = () => {
  //   setOpen(false);
  // };

  const navOpen = open ? "nav-open" : "";
  return (
    <div className={navOpen}>
      <header className={navOpen + " fixed-nav " + "l-header "}>
        <div className="l-header-inner">
          {/* <div className={styles.productLogo}>
            <PandasharkLogo width={128} height={64} />
          </div> */}

          {/* <!-- Navigation toggle button for Smartphone--> */}
          <button
            className={navOpen + " outside " + "nav-toggle-button "}
            data-about-nav-toggle
            aria-label="ナビゲーション メニューの切り替え"
            id="nav-toggle-button"
            onClick={() => {
              return toggleNav();
            }}
          >
            <img
              alt="ナビゲーション メニューを開く"
              className={navOpen + " menu-bars "}
              src="/img/icons/menu.png"
              height="24"
              width="24"
            />
            <img
              alt="ナビゲーション メニューを閉じる"
              className={navOpen + " menu-close "}
              src="/img/icons/close.png"
              height="24"
              width="24"
            />
          </button>
          {/* <HeaderMenus closeNav={closeNav} closeDialog={closeDialog} openDialog={openDialog} /> */}
        </div>
        <div>
          <div
            className="black-bg"
            role="button"
            id="js-black-bg"
            onKeyDown={(e) => {
              return closeNav(e);
            }}
            onClick={() => {
              return closeNav();
            }}
          ></div>
        </div>
      </header>
    </div>
  );
};
