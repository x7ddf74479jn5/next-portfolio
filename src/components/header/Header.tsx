import * as React from "react";
import PandasharkLogo from "src/components/common/PandasharkLogo";
import { HeaderMenus } from "src/components/header/HeaderMenus";
import Backdrop from "src/components/modal/Backdrop";
import useClickAway from "src/hooks/useClickAway";
import styles from "src/styles/components/header/Header.module.scss";

type Props = {
  isOpen: boolean;
  toggleNav(): void;
};

const NavToggleButton: React.VFC<Props> = ({ isOpen, toggleNav }) => {
  /* <!-- Navigation toggle button for Smartphone--> */

  const navToggleButtonStyle = isOpen ? styles.navToggleButtonOpen : styles.navToggleButton;

  return (
    <button
      className={navToggleButtonStyle}
      data-about-nav-toggle
      aria-label="ナビゲーション メニューの切り替え"
      id="nav-toggle-button"
      onClick={() => {
        return toggleNav();
      }}
    >
      {isOpen ? (
        <img alt="ナビゲーション メニューを閉じる" src="/img/icons/close.png" height="24" width="24" />
      ) : (
        <img alt="ナビゲーション メニューを開く" src="/img/icons/menu.png" height="24" width="24" />
      )}
    </button>
  );
};

export const Header: React.VFC = () => {
  const [isOpen, setIsOpen] = React.useState(false);
  const ref = React.useRef(null);

  const toggleNav = React.useCallback(() => {
    setIsOpen((prevState) => {
      return !prevState;
    });
  }, []);
  const closeNav = React.useCallback(() => {
    setIsOpen(false);
  }, []);

  useClickAway(ref, closeNav);

  return (
    <header className={styles.header}>
      <div className={styles.headerInner} ref={ref}>
        <div className={styles.productLogo}>
          <PandasharkLogo width={128} height={64} />
        </div>
        <NavToggleButton isOpen={isOpen} toggleNav={toggleNav} />

        <HeaderMenus closeNav={closeNav} isOpen={isOpen} />
      </div>
      {isOpen && <Backdrop />}
    </header>
  );
};
