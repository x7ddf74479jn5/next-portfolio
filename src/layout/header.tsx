import Image from "next/image";
import Link from "next/link";
import * as React from "react";
import { HeaderMenus } from "src/components/HeaderMenus";
import { images } from "src/static/links";

const items = [
  { href: "/", label: "Home" },
  { href: "/about", label: "About" },
];

type HeaderProps = {
  open: boolean;
  toggleNav(): void;
  closeNav(e?: React.KeyboardEvent<HTMLDivElement>): void;
};

export const Header: React.FC<HeaderProps> = (props) => {
  // const [open, setOpen] = React.useState(false);
  // const classes= {
  //   navOpen:"nav-open"
  // }
  // const toggleNav = () => {
  //   setOpen((prevState) => !prevState);
  // };
  // const closeNav = () => {
  //   setOpen(false);
  // };

  const navOpen = props.open ? "nav-open" : "";
  return (
    <div className={navOpen}>
      <header className={navOpen + " fixed-nav " + "l-header "}>
        {/* <h1>Title</h1>
    <nav>
      {items.map((item) => (
        <Link key={item.href} href={item.href}>
          <a style={{ display: "inline-block", padding: 12 }}>{item.label}</a>
        </Link>
      ))}
    </nav> */}

        <div className="l-header-inner">
          <h1 className="l-product-logo">
            <Link href="/">
              <a target="_self" aria-label="パンダシャーク">
                {/* <img alt="パンダシャーク" src="/img/icons/pandashark_logo.png" /> */}
                <Image width={128} height={64} alt="パンダシャーク" src={images.pandashark_rec} />
              </a>
            </Link>
          </h1>
          {/* <!-- Navigation toggle button for Smartphone--> */}
          <button
            className={navOpen + " outside " + "nav-toggle-button "}
            data-about-nav-toggle
            aria-label="ナビゲーション メニューの切り替え"
            id="nav-toggle-button"
            onClick={() => props.toggleNav()}
          >
            <img
              alt="ナビゲーション メニューを開く"
              className={navOpen + " menu-bars "}
              src="/img/icons/menu.png"
              height="24"
              width="24"
            />
            <img
              alt="ナビゲーション メニューを閉じる"
              className={navOpen + " menu-close "}
              src="/img/icons/close.png"
              height="24"
              width="24"
            />
          </button>
          <HeaderMenus />

          {/* <nav className="l-header__nav-top">
            <h1 className="l-product-logo">
              <a href="/" target="_self">
                <img alt="パンダシャーク" src="/img/icons/logo.png" />
              </a>
            </h1>
            <ul>
              <li>
                <a href="/about/" target="_self">
                  About
                </a>
              </li>
              <li>
                <a href="/samples/" target="_self">
                  Samples
                </a>
              </li>
              <li>
                <a href="/" target="_blank">
                  Blog
                </a>
              </li>
              <li>
                <a href="/contact/" target="_self">
                  Contact
                </a>
              </li>
              <li>
                <a className="sns" href="/" target="_self">
                  <img src="/img/icons/twitter_logo.png" alt="twitter" />
                  <span>Twitter</span>
                </a>
              </li>
            </ul>
          </nav>
           */}
        </div>
        <div>
          <div
            className="black-bg"
            role="button"
            id="js-black-bg"
            onKeyDown={(e) => props.closeNav(e)}
            onClick={() => props.closeNav()}
          ></div>
        </div>
      </header>
    </div>
  );
};
