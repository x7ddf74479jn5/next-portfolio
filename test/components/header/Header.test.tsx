import userEvent from "@testing-library/user-event";
import { Header } from "src/components/header/Header";
import type { RenderResult } from "test/test-utils";
import { render } from "test/test-utils";

describe("Header", () => {
  let renderResult: RenderResult;

  describe("closed", () => {
    beforeEach(() => {
      renderResult = render(<Header />);
    });

    it("matches snapshot", () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it("rendered correctly", () => {
      const navElement = renderResult.getByRole("banner");
      expect(navElement).toBeInTheDocument();
    });
    it("toggle header menu when SP", () => {
      const navToggleButton = renderResult.getAllByRole("button")[0];
      expect(navToggleButton).toHaveClass("navToggleButton");
      const nav = renderResult.getByRole("navigation");
      expect(nav).toHaveClass("headerNavTop");

      userEvent.click(navToggleButton);
      renderResult.rerender(<Header />);

      expect(nav).toHaveClass("headerNavTopOpen");
      expect(navToggleButton).toHaveClass("navToggleButtonOpen");

      userEvent.click(navToggleButton);
      renderResult.rerender(<Header />);

      expect(navToggleButton).toHaveClass("navToggleButton");
      expect(nav).toHaveClass("headerNavTop");
    });
  });
});
