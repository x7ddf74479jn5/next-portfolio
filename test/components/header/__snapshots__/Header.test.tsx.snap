// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header closed matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="headerInner"
    >
      <h1
        class="productLogo"
      >
        <a
          aria-label="パンダシャーク"
          href="/"
          target="_self"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTI4IiBoZWlnaHQ9IjY0IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <noscript />
            <img
              alt="Pandashark's icon"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </a>
      </h1>
      <button
        aria-label="ナビゲーション メニューの切り替え"
        class="navToggleButton"
        data-about-nav-toggle="true"
        id="nav-toggle-button"
      >
        <svg
          aria-hidden="true"
          aria-label="ナビゲーション メニューを開く\\""
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <nav
        class="headerNavTop"
      >
        <h1
          class="productLogo"
        >
          <a
            aria-label="パンダシャーク"
            href="/"
            target="_self"
          >
            <div
              style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="box-sizing: border-box; display: block; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  role="presentation"
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iODAiIGhlaWdodD0iNDAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                  style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                />
              </div>
              <noscript />
              <img
                alt="Pandashark's icon"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </a>
        </h1>
        <ul>
          <li>
            <a
              class="linkItem"
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              class="linkItem"
              href="/samples"
            >
              Samples
            </a>
          </li>
          <li>
            <a
              class="linkItem"
              href="/contact"
            >
              Contact
            </a>
          </li>
          <li>
            <a
              class="linkItem"
              href="https://twitter.com/pandashark6"
            >
              <img
                alt="twitter"
                src="/img/icons/twitter-logo.svg"
              />
              <span>
                Twitter
              </span>
            </a>
          </li>
          <li>
            <a
              class="linkItem"
              href="https://github.com/x7ddf74479jn5"
            >
              <img
                alt="github"
                src="/img/icons/github-logo.svg"
              />
              <span>
                GitHub
              </span>
            </a>
          </li>
          <li>
            <button
              class="buttonItem"
            >
              <img
                alt="chatbot"
                src="/img/icons/question_answer_black.svg"
              />
              <span>
                Chatbot
              </span>
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</DocumentFragment>
`;
